examples/fabric-admin/rpc/RpcClient.cpp

```diff
@@ -0,0 +1,168 @@
+#include <unistd.h>
+
+#include "fabric_bridge_service/fabric_bridge_service.rpc.pb.h"
+#include "pw_assert/check.h"
+#include "pw_hdlc/decoder.h"
+#include "pw_hdlc/default_addresses.h"
+#include "pw_hdlc/rpc_channel.h"
+#include "pw_rpc/client.h"
+#include "pw_stream/socket_stream.h"
+
+namespace {
+
+constexpr size_t kMaxTransmissionUnit = 256;
+constexpr uint32_t kRpcTimeoutMs      = 1000;
+const char * rpcServerAddress         = "127.0.0.1";
+
+pw::stream::SocketStream rpcSocketStream;
+
+// Set up the output channel for the pw_rpc client to use.
+pw::hdlc::RpcChannelOutput hdlc_channel_output(rpcSocketStream, pw::hdlc::kDefaultRpcAddress, "HDLC channel");
+
+// An array of RPC channels (channels) is created, each associated with an HDLC channel output.
+// This sets up the communication channels for RPC calls.
+pw::rpc::Channel channels[] = { pw::rpc::Channel::Create<1>(&hdlc_channel_output) };
+
+// Initialize the RPC client with the channels.
+pw::rpc::Client client(channels);
+
+// Generated clients are namespaced with their proto library.
+using FabricBridgeClient = chip::rpc::pw_rpc::nanopb::FabricBridge::Client;
+
+// RPC channel ID on which to make client calls. RPC calls cannot be made on
+// channel 0 (Channel::kUnassignedChannelId).
+constexpr uint32_t kDefaultChannelId = 1;
+
+// Function to process incoming packets
+void ProcessPackets()
+{
+    std::array<std::byte, kMaxTransmissionUnit> inputBuf;
+    pw::hdlc::Decoder decoder(inputBuf);
+
+    while (true)
+    {
+        std::array<std::byte, kMaxTransmissionUnit> data;
+        auto ret = rpcSocketStream.Read(data);
+        if (!ret.ok())
+        {
+            if (ret.status() == pw::Status::OutOfRange())
+            {
+                // Handle remote disconnect
+                rpcSocketStream.Close();
+                return;
+            }
+            continue;
+        }
+
+        for (std::byte byte : ret.value())
+        {
+            auto result = decoder.Process(byte);
+            if (!result.ok())
+            {
+                // Wait for more bytes that form a complete packet
+                continue;
+            }
+            pw::hdlc::Frame & frame = result.value();
+            if (frame.address() != pw::hdlc::kDefaultRpcAddress)
+            {
+                // Wrong address; ignore the packet
+                continue;
+            }
+
+            client.ProcessPacket(frame.data()).IgnoreError();
+        }
+    }
+}
+
+template <typename CallType>
+CHIP_ERROR WaitForResponse(CallType & call)
+{
+    if (!call.active())
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    // Wait for the response or timeout
+    uint32_t elapsedTimeMs     = 0;
+    const uint32_t sleepTimeMs = 100;
+
+    while (call.active() && elapsedTimeMs < kRpcTimeoutMs)
+    {
+        usleep(sleepTimeMs * 1000);
+        elapsedTimeMs += sleepTimeMs;
+    }
+
+    if (elapsedTimeMs >= kRpcTimeoutMs)
+    {
+        fprintf(stderr, "RPC Response timed out!");
```

Comment:
Please do not user straight fprintf, use ChipLogError.

Reply:
We need to print out this message to console instead of log